community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -482,10 +477,9 @@ public NodePropertyExistenceConstraint nodePropertyExistenceConstraintCreate( Ke
     public RelationshipPropertyExistenceConstraint relationshipPropertyExistenceConstraintCreate( KernelStatement state,
             int relTypeId, int propertyKeyId ) throws AlreadyConstrainedException, CreateConstraintFailureException
     {
-        try ( Cursor<RelationshipItem> cursor = relationshipCursorGetAll( state ) )
-        {
-            constraintSemantics.validateRelationshipPropertyExistenceConstraint( cursor, relTypeId, propertyKeyId );
-        }
+        Iterator<Cursor<RelationshipItem>> relationships = new EntityLoadingIterator<>( relationshipsGetAll( state ),
+                (id) -> relationshipCursorById( state, id ) );
+        constraintSemantics.validateRelationshipPropertyExistenceConstraint( relationships, relTypeId, propertyKeyId );
```

Comment:
the relationship cursor was loading the record twice as well, there is no change there. See https://github.com/neo4j/neo4j/blob/3.2/community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreStatement.java#L160

I agree the `Iterator<Cursor<...>>` is silly, I plan to clean that up later (see the other comment).